code/modules/mob/living/carbon/xenomorph/abilities/queen/queen_powers.dm

```diff
@@ -383,9 +390,116 @@
 		if("De-evolve (500)")
 			de_evolve_other()
 		if("Reward Jelly (500)")
-			give_jelly_reward(queenbanish.hive)
+			give_jelly_reward(queen_manager.hive)
 		if("Exchange larva for evolution (100)")
 			give_evo_points()
+		if("Add Personal Ally")
+			add_personal_ally()
+		if("Remove Personal Ally")
+			remove_personal_ally()
+		if("Clear Personal Allies")
+			clear_personal_allies()
+
+/datum/action/xeno_action/onclick/manage_hive/proc/add_personal_ally()
+	var/mob/living/carbon/xenomorph/queen/user_xeno = owner
+	if(user_xeno.hive.hivenumber != XENO_HIVE_CORRUPTED)
+		return
+
+	if(!user_xeno.check_state())
+		return
+
+	var/datum/hive_status/corrupted/hive = user_xeno.hive
+	var/list/target_list = list()
+	for(var/mob/living/carbon/human/possible_target in view(7, user_xeno))
```

Comment:
Is user_xeno here always the queen even on ovi and in eye view - or is it where she is looking?  Ideally this should be in an area the queen is looking.

Reply:
Does work now (without ovi, on ovi and in eye form away from the queen itself). I wonder if I need some additional checks here (like if eye is actually a queen eye), not sure if I am missing anything.

Also I didn't know that view checks for walls even if you have wallhack so I use range now.

![2024-04-17_22-37-10](https://github.com/cmss13-devs/cmss13/assets/115417687/fc6caa36-3d15-43ab-9243-fcf139b4b7e7)